### ISSUE TITLE
[Bug] The column type "model_function" does not use the limit value defined

### ISSUE THREAD

**Issue by DanielKimmich** [Laravel-Backpack/CRUD#2959]
# Bug report

### What I did

```
        $this->crud->addColumn([
            'name' => 'url',
            'label' => ucfirst(trans('backpack::permissionmanager.permission_plural')),
            'type' => 'model_function',
            'function_name' => 'PermissionsViaRoles', // the method in your Model
           'limit' => 1500, // Limit the number of characters shown
            ]);

```

### What I expected to happen

show everything

### What happened

truncated at 999 characters

### What I've already tried to fix it

```
@php
    $value = $entry->{$column['function_name']}(...($column['function_parameters'] ?? []));
    dump($value);
    $column['escaped'] = $column['escaped'] ?? false;
    $column['limit']   = $column['limit'] ?? 40;
    $column['prefix']  = $column['prefix'] ?? '';
    $column['suffix']  = $column['suffix'] ?? '';
    $column['text']    = $column['prefix'].
                         Str::limit($value, $column['limit'], "[...]").
                         $column['suffix'];
dump($column['limit']);
dump($column['text']);
@endphp
```

                                                                                                                               
"list authrole, show authrole, create authrole, update authrole, delete authrole, list authuser, show authuser, create authuser, update authuser, delete authuser, list authpermission, show authpermission, create authpermission, update authpermission, delete authpermission, list worldcontinent, show worldcontinent, create worldcontinent, update worldcontinent, delete worldcontinent, list worldcountry, show worldcountry, create worldcountry, update worldcountry, delete worldcountry, list worlddivision, show worlddivision, create worlddivision, update worlddivision, delete worlddivision, list worldcity, show worldcity, create worldcity, update worldcity, delete worldcity, list managerbackup, list managerlog, list managersetting, list managermigrate, list contactperson, show contactperson, create contactperson, update contactperson, delete contactperson, list contactfamily, show contactfamily, create contactfamily, update contactfamily, delete contactfamily, list contactsetting, show contactsetting, create contactsetting, update contactsetting, delete contactsetting, list blogpost, show blogpost, create blogpost, update blogpost, delete blogpost, list blogcomment, show blogcomment, create blogcomment, update blogcomment, delete blogcomment, list blogcategory, show blogcategory, create blogcategory, update blogcategory, delete blogcategory, list blogtag, show blogtag, create blogtag, update blogtag, delete blogtag"                                                                                           
                                                                                                                               
999                                                                                                                            
                                                                                                                               
"list authrole, show authrole, create authrole, update authrole, delete authrole, list authuser, show authuser, create authuser, update authuser, delete authuser, list authpermission, show authpermission, create authpermission, update authpermission, delete authpermission, list worldcontinent, show worldcontinent, create worldcontinent, update worldcontinent, delete worldcontinent, list worldcountry, show worldcountry, create worldcountry, update worldcountry, delete worldcountry, list worlddivision, show worlddivision, create worlddivision, update worlddivision, delete worlddivision, list worldcity, show worldcity, create worldcity, update worldcity, delete worldcity, list managerbackup, list managerlog, list managersetting, list managermigrate, list contactperson, show contactperson, create contactperson, update contactperson, delete contactperson, list contactfamily, show contactfamily, create contactfamily, update contactfamily, delete contactfamily, list contactsetting, show conta[...]"          


### Backpack, Laravel, PHP, DB version

### PHP VERSION:
PHP 7.2.11 (cli) (built: Oct 10 2018 02:04:07) ( ZTS MSVC15 (Visual C++ 2017) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

### LARAVEL VERSION:
v7.16.1@dc9cd8338d222dec2d9962553639e08c4585fa5b

### BACKPACK VERSION:
4.1.10@cf33f991b97de99cefa1dcd2e6beeeeda8116514

**Comment by pxpm** [comment]
Hello @DanielKimmich 

I was not able to reproduce it, it could be something with environment I think. I am in 4.1.11, but there is no changes here.

```php
 [
                // show both text and email values in one column
                // this column is here to demo and test the custom searchLogic functionality
                'name'          => 'model_function',
                'label'         => 'Text and Email', // Table column heading
                'type'          => 'model_function',
                'function_name' => 'getTextAndEmailAttribute', // the method in your Model
                'limit' => 1500,
            ],
```
` dd($column['limit']);` in `model_function.blade.php` is `1500`. 

Can you provide me a way to reproduce it ?

Best,
Pedro

**Comment by DanielKimmich** [comment]
What you could do to reproduce it is to load 10 email records that have a size of 120 characters, that sum gives 1200 characters, if it does not show everything, because it would be limiting it.

I hope that serves to validate.

**Comment by DanielKimmich** [comment]
In this function "Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation.php" is where the change is made

```
            // remove the character limit on columns that take it into account
            if (in_array($column['type'], ['text', 'email', 'model_function', 'model_function_attribute', 'phone', 'row_number', 'select'])) {
                $this->crud->modifyColumn($column['key'], ['limit' => 999]);
            }
```

What would be the reason?

**Comment by pxpm** [comment]
Thanks @DanielKimmich for pointing me out in the correct direction, I was not looking into ShowOperation.

I think that this is assuming that these types of columns have their limit setup by us by default for the ListOperation and we override them to be bigger on show. 

I agree that if developer sets the limit he knows what he is doing, so I proposed the PR #2968 

If you'd like to have a different column limit in you list operation just use the same technique on `setupListOperation` using `$this->crud->modifyColumn()`.

I will be closing this, let's keep any further discussion on this regard in the PR thread. Thanks again @DanielKimmich :+1: :1st_place_medal: 

Best,
Pedro

**Comment by DanielKimmich** [comment]
Thank you

**Comment by tabacitu** [comment]
Totally! Thanks @DanielKimmich , I've just merged @pxpm 's PR ! It will be tagged and available starting with 4.1.12 later today.
